docs/Antora/modules/howto_guides/pages/howto_measure_single_value.adoc

```diff
@@ -1,33 +1,305 @@
 
-OpenDAQ offers a variety of solutions for reading signal values, including readers. For developers requiring access to only the most recent value, the signal interface provides a method named "getLastValue."
+For developers requiring access to only the most recent value, the Signal and Data Packet interfaces provide a method named `getLastValue` (`last_value` in Python).
+
+[#last_value_signal]
+== Getting the last value of a Signal
+
+When the `getLastValue` method (`last_value` in Python)  is called on a Signal, it yields an `IBaseObject` containing the last value.
+
+[tabs]
+====
+Cpp::
++
+[source,cpp]
+----
+// Get last value of a Signal
+auto lastValue = signal.getLastValue();
+----
+Python::
++
+[source,python]
+----
+
+----
+====
+
+[#last_value_data_packet]
+== Getting last value of a Data Packet
+
+Similar is true when calling the `getLastValue` method (`last_value` in Python)  on a Data Packet.
+
+[tabs]
+====
+Cpp::
++
+[source,cpp]
+----
+// Get last value of a Data Packet
+auto lastValue = packet.getLastValue();
+----
+Python::
++
+[source,python]
+----
+
+----
+====
+
+[#printing_the_last_value]
+== Printing the last value
+
+No matter the source (either the Signal or the Data Packet), the resulting variable can be, for example, printed into the console.
+
+[tabs]
+====
+Cpp::
++
+[source,cpp]
+----
+// Print last value
+std::cout << "lastValue: " << lastValue << std::endl;
+----
+Python::
++
+[source,python]
+----
+
+----
+====
+
+[#casting_the_last_value]
+== Casting the last value and using the cast variable
+
+In C++, the function `getLastValue` returns an object of type `IBaseObject`. To access it's underying members and methods, one must explicitly cast it into the appropriate type. However, in Python, this is not necessary due to dynamic typing.
+
+The underlying type of the `IBaseObject` we get from our `getLastValue` call can be cast into either:
+
+* `IFloat` if the type is `Float32` or `Float64`
+* `IInteger` if the type is `Int8` through `Int64` or `UInt8` through `UInt64`
+* `IComplexNumber` if the type is `ComplexFloat32` or `ComplexFloat64`
+* `IRange` if the type is `RangeInt64`
+* `IStruct` if the type is `Struct`
+* `IList` of the aforementioned types if there is exactly one dimension in the single value
+
+An example of casting the `lastValue` object into an `IFloat` object:
+
+[tabs]
+====
+Cpp::
++
+[source,cpp]
+----
+// Cast to IFloat
+auto number = lastValue.asPtr<IFloat>();
+----
+Python::
++
+[source,python]
+----
+
+----
+====
+
+In the above example, the Data Descriptor of the Data Packet or its Signal on which `getLastValue` (`last_value` in Python) is called might be:
+
+[tabs]
+====
+Cpp::
++
+[source,cpp]
+----
+// The Data Descriptor for Float64
+auto descriptor = DataDescriptorBuilder().setSampleType(SampleType::Float64).build();
+----
+Python::
++
+[source,python]
+----
+
+----
+====
+
+[#last_value_range]
+== Getting the last value with a Range
```

Comment:
I'd omit this section. We just need a list of type conversions at the top for users to refer to. Beyond that they should understand how to use certain openDAQ types - this should be a separate guide if we need it.